app/code/core/Mage/Adminhtml/controllers/Customer/Address/AttributeController.php

```diff
@@ -0,0 +1,328 @@
+                Mage::getSingleton('adminhtml/session')->addError(
+                    Mage::helper('customer')->__('This attribute no longer exists'));
+                $this->_redirect('*/*/');
+                return;
+            }
+
+            // entity type check
+            if ($model->getEntityTypeId() != $this->_entityTypeId) {
+                Mage::getSingleton('adminhtml/session')->addError(
+                    Mage::helper('customer')->__('This attribute cannot be edited.'));
+                $this->_redirect('*/*/');
+                return;
+            }
+        }
+
+        // set entered data if was error when we do save
+        $data = Mage::getSingleton('adminhtml/session')->getAttributeData(true);
+        if (!empty($data)) {
+            $model->addData($data);
+        }
+
+        Mage::register('entity_attribute', $model);
+        $this->_initAction();
+        $this->_title($id ? $model->getName() : $this->__('New Attribute'));
+        $item = $id ? Mage::helper('customer')->__('Edit Customer Attribute')
+            : Mage::helper('customer')->__('New Customer Attribute');
+        $this->_addBreadcrumb($item, $item);
+        $this->renderLayout();
+    }
+
+    public function validateAction()
+    {
+        $response = new Varien_Object();
+        $response->setError(false);
+
+        $attributeCode  = $this->getRequest()->getParam('attribute_code');
+        $attributeId    = $this->getRequest()->getParam('attribute_id');
+        $attribute = Mage::getModel('customer/attribute')
+            ->loadByCode($this->_entityTypeId, $attributeCode);
+
+        if ($attribute->getId() && !$attributeId) {
+            Mage::getSingleton('adminhtml/session')->addError(
+                Mage::helper('customer')->__('Attribute with the same code already exists'));
+            $this->_initLayoutMessages('adminhtml/session');
+            $response->setError(true);
+            $response->setMessage($this->getLayout()->getMessagesBlock()->getGroupedHtml());
+        }
+
+        $this->getResponse()->setBody($response->toJson());
+    }
+
+    /**
+     * Filter post data
+     *
+     * @param array $data
+     * @return array
+     */
+    protected function _filterPostData($data)
+    {
+        if ($data) {
+            /** @var Mage_Customer_Helper_Data $helper */
+            $helper = Mage::helper('customer');
+            //labels
+            $data['frontend_label'] = (array) $data['frontend_label'];
+            foreach ($data['frontend_label'] as & $value) {
+                if ($value) {
+                    $value = $helper->stripTags($value);
+                }
+            }
+
+            if (!empty($data['option']) && !empty($data['option']['value']) && is_array($data['option']['value'])) {
+                $allowableTags = isset($data['is_html_allowed_on_front']) && $data['is_html_allowed_on_front']
+                    ? sprintf('<%s>', implode('><', $this->_getAllowedTags())) : null;
```

Comment:
`_getAllowedTags` is missing in this controller:

```php
/**
 * Get list of allowed text formatted as array
 *
 * @return array
 */
protected function _getAllowedTags()
{
    return explode(',', Mage::getStoreConfig(self::XML_PATH_ALLOWED_TAGS));
}
```

Reply:
ok I've added the method, but it uses a constant that's `const XML_PATH_ALLOWED_TAGS = 'system/catalog/frontend/allowed_html_tags_list';` so what should we do with that? change it to `system/customer/frontend/allowed_html_tags_list`?

Reply:
I'm not sure if we ever need that for the customer attributes, I don't think any of them allow html. I'll have to look into that.